packages/generator/src/parser/edmx-parser-v4.ts

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import { forceArray } from '../generator-utils';
+import {
+  EdmxEntitySet,
+  EdmxMetadata,
+  EdmxEntityType,
+  EdmxEnumType,
+  EdmxDerivedType,
+  EdmxComplexType
+} from './parser-types-v4';
+import {
+  parseEntityTypes,
+  parseFunctionImports,
+  parseComplexTypes
+} from './edmx-parser-common';
+import { stripNamespace, parseTypeName } from './parser-util';
+import { EdmxMetadataBase } from './parser-types-common';
+
+function joinTypesWithBaseTypes<T extends EdmxDerivedType>(
+  types: T[],
+  joinTypes: (type: T, baseType: T) => T
+) {
+  return types.map(type =>
+    type.BaseType ? addBaseTypeToType(type, types, joinTypes) : type
+  );
+}
+
+function addBaseTypeToType<T extends EdmxDerivedType>(
+  type: T,
+  types: T[],
+  joinTypes: (type: T, baseType: T) => T
+) {
+  const baseType = types.find(e => e.Name === stripNamespace(type.BaseType!));
+
+  if (!baseType) {
+    throw new Error(
+      `Type ${type.BaseType} not found, but defined as BaseType of Type ${type.Name}.`
+    );
+  }
+
+  if (baseType.BaseType) {
+    return joinTypes(type, addBaseTypeToType(baseType, types, joinTypes));
```

Comment:
recursion always breaks my head. Could you perhaps add a little JS doc which explains what this message does.

Reply:
done